index.js

```diff
@@ -23,50 +23,80 @@ function getMetadata (opts, dir, cb) {
   var props = []
   if (!opts.name) props.push(['productName', 'name'])
   if (!opts['app-version']) props.push('version')
-  if (!opts.version) props.push(['dependencies.electron', 'devDependencies.electron'])
+  if (!opts.version) {
+    props.push([
+      'dependencies.electron',
+      'devDependencies.electron',
+      'dependencies.electron-prebuilt',
+      'devDependencies.electron-prebuilt'
+    ])
+  }
 
   // Name and version provided, no need to infer
   if (props.length === 0) return cb(null)
 
   // Search package.json files to infer name and version from
-  getPackageInfo(props, dir, function (err, result) {
-    if (err) {
-      // `get-package-info` exploded looking for `electron`. Try `electron-prebuilt` instead
-      props.pop()
-      props.push(['dependencies.electron-prebuilt', 'devDependencies.electron-prebuilt'])
-      getPackageInfo(props, dir, function (err, result) {
-        if (err) return cb(err)
-        return inferNameAndVersionFromInstalled('electron-prebuilt', opts, result, cb)
+  getPackageInfo(props, dir, (err, result) => {
+    if (err && err.missingProps) {
+      let requiredProps = ['productName', 'dependencies.electron']
+      let missingProps = err.missingProps.map(prop => {
+        return Array.isArray(prop) ? prop[0] : prop
       })
-    } else {
-      return inferNameAndVersionFromInstalled('electron', opts, result, cb)
+
+      // Callback w/ error if there are required props that are missing
+      if (missingProps.filter(prop => requiredProps.find(reqProp => prop === reqProp)).length !== 0) {
+        let messages = missingProps.map(missingProp => {
+          let hash, propName
+          if (missingProp === 'productName') {
+            hash = 'name'
+            propName = 'application name'
+          }
+
+          if (missingProp === 'dependencies.electron') {
+            hash = 'version'
+            propName = 'Electron version'
+          }
+
+          return `Unable to determine ${propName}. Please specify an ${propName}\n\n` +
+            'For more information, please see\n' +
+            `https://github.com/electron-userland/electron-packager/blob/master/docs/api.md#${hash}\n`
+        })
+
+        debug(err.message)
+        err.message = messages.join('\n') + '\n'
+        return cb(err)
+      } else {
+        // Missing props not required, can continue w/ partial result
+        result = err.result
+      }
+    } else if (err) {
+      return cb(err)
```

Comment:
Would an IO error include an invalid `package.json`? Just trying to think of ways to test this line.